flax/linen/module.py

```diff
@@ -379,13 +379,34 @@ def _get_local_method_names(cls: Any,
     A list of method names.
   """
   true_methods = set()
-  for m in cls.__dict__:
-    if callable(cls.__dict__[m]) and not inspect.isclass(cls.__dict__[m]):
-      mtype = type(cls.__dict__[m])
+  for m, attr in cls.__dict__.items():
+    if callable(attr) and not inspect.isclass(attr):
+      mtype = type(attr)
       if mtype != staticmethod and mtype != classmethod:
         true_methods.add(m)
   return tuple(true_methods.difference(set(exclude)))
 
+def _get_local_descriptor_names(cls: Any,
+                                exclude: Iterable[str] = ()) -> Tuple[str, ...]:
+  """Gets descriptor names of a class.
+
+  Args:
+    cls: The class to get property names for.
+    exclude: Names to exclude from output.
+  Returns:
+    A list of property names.
+  """
+  true_properties = set()
+  for m, attr in cls.__dict__.items():
+    if not callable(attr) and (
+      hasattr(attr, '__get__') or hasattr(attr, '__set__') or
+      hasattr(attr, '__delete__')
+    ):
+      mtype = type(attr)
+      if mtype != staticmethod and mtype != classmethod:
```

Comment:
is there a need to check for `staticmethod`/`classmethod` if the attr isn't a callable??